File: etalon/RBXCalculatorRoboxCounter2 – копія/app/src/main/java/com/robuxe/robuxtracker/freerobux/adsmodule/adscode/ADPref.java
package com.robuxe.robuxtracker.freerobux.adsmodule.adscode;

import android.content.Context;
import android.content.SharedPreferences;

import com.robuxe.robuxtracker.freerobux.adsmodule.model.RemoteConfigModel;
import com.google.gson.Gson;

import kotlin.jvm.internal.DefaultConstructorMarker;
import kotlin.jvm.internal.Intrinsics;

public final class ADPref {
    public static final Companion INSTANCE = new Companion(null);
    private static final String CONFIG_KEY = "AdsDATAConfig";
    private static volatile ADPref instance;
    private static SharedPreferences sharedPreferences;

    public ADPref(Context context, DefaultConstructorMarker defaultConstructorMarker) {
        this(context);
    }

    private ADPref(Context context) {
        SharedPreferences sharedPreferences2 = context.getSharedPreferences("ff_maxskintool", 0);
        Intrinsics.checkNotNullExpressionValue(sharedPreferences2, "getSharedPreferences(...)");
        sharedPreferences = sharedPreferences2;
    }

    public final void saveRemoteConfig(RemoteConfigModel remoteConfigModel) {
        Intrinsics.checkNotNullParameter(remoteConfigModel, "remoteConfigModel");
        SharedPreferences sharedPreferences2 = sharedPreferences;
        if (sharedPreferences2 == null) {
            Intrinsics.throwUninitializedPropertyAccessException("sharedPreferences");
            sharedPreferences2 = null;
        }
        SharedPreferences.Editor editor = sharedPreferences2.edit();
        Gson gson = new Gson();
        String json = gson.toJson(remoteConfigModel);
        editor.putString(CONFIG_KEY, json);
        editor.apply();
    }

    public final RemoteConfigModel getRemoteConfig() {
        Gson gson = new Gson();
        SharedPreferences sharedPreferences2 = sharedPreferences;
        if (sharedPreferences2 == null) {
            Intrinsics.throwUninitializedPropertyAccessException("sharedPreferences");
            sharedPreferences2 = null;
        }
        String json = sharedPreferences2.getString(CONFIG_KEY, null);
        return (RemoteConfigModel) gson.fromJson(json, RemoteConfigModel.class);
    }

    public static final class Companion {
        public Companion(DefaultConstructorMarker defaultConstructorMarker) {
            this();
        }

        private Companion() {
        }

        public final ADPref getInstance(Context context) {
            ADPref it;
            Intrinsics.checkNotNullParameter(context, "context");
            ADPref aDPref = ADPref.instance;
            if (aDPref != null) {
                return aDPref;
            }
            synchronized (this) {
                it = ADPref.instance;
                if (it == null) {
                    it = new ADPref(context, null);
                    Companion companion = ADPref.INSTANCE;
                    ADPref.instance = it;
                }
            }
            return it;
        }
    }

}

File: april_may/Game T1/T1 with ADS/app/src/main/java/com/rbuxrds/counterds/adsmodule/AdPref.kt
package com.rbuxrds.counterds.adsmodule

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.Gson
import androidx.core.content.edit

// Зберігає дані реклами між запусками додатку
// Щоб при наступному запуску реклама одразу працювала
// без очікування відповіді від Firebase
class AdPref(context: Context) {

    private val prefs: SharedPreferences = context.getSharedPreferences("rbux_ads_prefs", Context.MODE_PRIVATE)

    private val gson = Gson()

    // ── RemoteConfig ──────────────────────────────────────────────────────────
    // Зберігаємо весь конфіг як JSON рядок

    fun saveConfig(model: RemoteConfigModel) {
        prefs.edit { putString(KEY_CONFIG, gson.toJson(model)) }
    }

    fun loadConfig(): RemoteConfigModel? {
        val json = prefs.getString(KEY_CONFIG, null) ?: return null
        return runCatching { gson.fromJson(json, RemoteConfigModel::class.java) }.getOrNull()
    }

    // ── UserType ──────────────────────────────────────────────────────────────
    // Зберігаємо тип юзера — визначається один раз і більше не змінюється

    fun saveUserType(type: UserType) {
        prefs.edit { putString(KEY_USER_TYPE, type.name) }
    }

    fun loadUserType(): UserType? {
        val name = prefs.getString(KEY_USER_TYPE, null) ?: return null
        return runCatching { UserType.valueOf(name) }.getOrNull()
    }

    // ── Navigation counters ───────────────────────────────────────────────────
    // Лічильник front навігації — зберігається між сесіями
    // (коли reset_on_app_restart = false)

    var frontNavCount: Int
        get() = prefs.getInt(KEY_FRONT_NAV, 0)
        set(value) = prefs.edit { putInt(KEY_FRONT_NAV, value) }

    fun resetFrontNav() {
        frontNavCount = 0
    }

    companion object {
        private const val KEY_CONFIG    = "remote_config_json"
        private const val KEY_USER_TYPE = "user_type"
        private const val KEY_FRONT_NAV = "front_nav_count"
    }
}